src/Compilers/VisualBasic/Portable/Lowering/LocalRewriter/LocalRewriter_Conversion.vb

```diff
@@ -96,10 +96,24 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             End If
 
             If node.RelaxationLambdaOpt IsNot Nothing Then
-                returnValue = node.Update(VisitExpressionNode(node.RelaxationLambdaOpt),
+
+                If _inExpressionLambda AndAlso
+                    NoParameterRelaxation(DirectCast(node.Operand, BoundLambda).LambdaSymbol, node.RelaxationLambdaOpt.LambdaSymbol) Then
```

Comment:
What prevents you from using (not declaring) the local inside the lambda that is converted to an expression tree?

---
In reply to: [64994396](https://github.com/dotnet/roslyn/pull/11627#discussion_r64994396) [](ancestors = 64994396)